content/blog/new-graphic-features/index.Rmd

```diff
@@ -0,0 +1,301 @@
+
+Gradients can be created with the `linearGradient()` and `radialGradient()` which can be assigned to the fill of a grob:
+
+```{r}
+library(grid)
+grid.circle(
+  gp = gpar(
+    fill = linearGradient(
+      c("firebrick", "steelblue", "forestgreen"),
+      stops = c(0, 0.7, 1)
+    ),
+    col = NA
+  )
+)
+```
+
+```{r}
+grid.circle(
+  gp = gpar(
+    fill = radialGradient(
+      c("white", "steelblue"),
+      cx1 = 0.8,
+      cy1 = 0.8
+    ),
+    col = NA
+  )
+)
+```
+
+At the basic level, both of the constructors takes a vector of colors. Optionally you can provide a vector of stops that define where along the span of the gradient each color is placed. Each gradient type also lets you specify where in the graphic the gradient runs between. For a linear gradient you provide the x and y position of the start and end of the gradient, for a radial gradient you provide the center and radius of the start and end circle. Lastly, you can also tell it how it should behave outside of the given range using the `extend` argument:
+
+```{r}
+grid.circle(
+  gp = gpar(
+    fill = linearGradient(
+      c("firebrick", "steelblue"),
+      x1 = 0, y1 = 0, x2 = 0.5, y2 = 0,
+      extend = "repeat"
+    ),
+    col = NA
+  )
+)
+```
+
+```{r}
+grid.circle(
+  gp = gpar(
+    fill = linearGradient(
+      c("firebrick", "steelblue"),
+      x1 = 0, y1 = 0, x2 = 0.5, y2 = 0,
+      extend = "pad"
+    ),
+    col = NA
+  )
+)
+```
+
+One thing to note is that the extend of the gradient is given relative to the bounding box of the grob being drawn. We could move the circle above around and the gradient would follow along with it.
+
+### Patterns
+
+Like gradients, patterns are a new type of fill made possible in grid through the new features in the graphic engine. Patterns are crazy powerful in that they can be *any* grob you can imagine. The grob itself can consist of other grobs and these grobs could have patterned fill as well (or gradient fills for that matter).
+
+```{r}
+gradient_rec <- rectGrob(
+  width = 0.1, 
+  height = 0.1, 
+  gp = gpar(fill = linearGradient(c("firebrick", "steelblue")))
+)
+grid.draw(gradient_rec)
+```
+
+```{r}
+grid.circle(
+  gp = gpar(fill = pattern(gradient_rec, width = 0.15, height = 0.15, extend = "reflect"))
+)
+```
+
+Understanding the sizing of the grob used for the pattern can take some getting used to. Basically the pattern is drawn relative to the grob. The `width` and `height` argument in the `pattern()` call is then used to define a region of the grob that will be used as a pattern. You can thus not scale the pattern grob using the `width` and `height` argument in `pattern()`. This can be seen below
```

Comment:
```suggestion
Understanding the sizing of the grob used for the pattern can take some getting used to. Basically, the pattern is drawn relative to the grob. The `width` and `height` arguments in the `pattern()` call is then used to define a region of the grob that will be used as a pattern. Thus, you cannot scale the pattern grob using the `width` and `height` arguments in `pattern()`. This can be seen below
```